### ISSUE TITLE
`monaco-graphql` monaco import bundles all basic languages

### ISSUE THREAD

**Issue by lokmeinmatz** [graphql/graphiql#2672]
The `monaco-graphql` package has the following import at multiple places:
`import * as monaco from 'monaco-editor'`

This import loads `monaco-editor/esm/vs/editor/editor.main.js` which in turn loads `../basic-languages/monaco.contribution` which contains many of the "basic languages" supported by monaco.

When importing this language plugin, also only `initializeMode` like shown in the examples, all these languages are bundled. (see this minimal example using vite: https://github.com/lokmeinmatz/monaco-graphql-lang-bundles , the stats.html is generated and shows the import dependencies).

![grafik](https://user-images.githubusercontent.com/21058206/184557526-f590cfef-7c2c-4252-bc25-400ab77f4cac.png)

I don't have the time or expertise to create a Pull request, but is it possible to change all these imports to
`import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';`?

Thanks, Matthias

**Comment by acao** [comment]
Unfortunately this is not possible because we can’t import individual monaco.languages that we need like json. This isn’t as simple as changing the worker imports as you suggest, which we switched away from because it didn’t have any impact.

A PR would be welcome though! Perhaps they have changed the way languages export. But in order to use monaco.languages.json‘s API we have to import languages, and last I checked last year at least there was no way to import just the loaded json monaco mode
